src/tools/compiletest/tidy-rustdoc.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+indir="${1:?Missing argument 1: input directory}"
+
+tidy () {
+  {
+    # new-inline-tags is workaround for:
+    #   https://github.com/rust-lang/stdarch/issues/945
+    #   https://github.com/rust-lang/mdBook/issues/1372
+    command tidy \
+        --indent yes \
+        --indent-spaces 2 \
+        --wrap 0 \
+        --show-warnings no \
+        --markup yes \
+        --quiet yes \
+        --new-inline-tags 'c t' \
+        "$@" \
+        >/dev/null \
+    || {
+      # tidy exits with code 1 if there were any warnings :facepalm:
+      status=$?
+      if [ $status != 0 ] && [ $status != 1 ]
+      then
+        echo "While tidying $1" >&2
+        exit 1
+      fi
+    }
```

Comment:
```suggestion
    || test $? -eq 1 \
    || { # ^ tidy exits with code 1 if there were any warnings, ignore it 🤦
      echo "While tidying $1" >&2
      exit 1
    }
```

I don't think we need to check `0` if it is on the `||` branch since `0` already indicates success.

Heh, and a real facepalm.